zngur-generator/src/cpp.rs

```diff
@@ -1658,13 +1658,25 @@ namespace rust {
     struct RefMut;
 
     template<typename T, size_t OFFSET>
-    struct FieldOwned;
+    struct FieldOwned {
+        inline ::rust::Ref<T> as_ref() const noexcept { return ::rust::Ref<T>(*this); }
+        inline T read() const noexcept { auto r = as_ref(); return *r; }
```

Comment:
I like the `operator T()` but I'm not sure about `.read()` and `.as_ref()`. They may have name collisions with the methods on `T` (specially `.as_ref` which is a method on `Option<T>` and many other types in the std) and they are not needed in the Rust syntax. The user can do `T(some.field)` instead of `some.field.read()` and `Ref<T>(some.field)` for `some.field.as_ref()`, so do these functions pull their weight?